src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesHttpClient.java

```diff
@@ -78,6 +83,26 @@ public Response delete(String resourceEndpoint, String appId) {
 		return response;
 	}
 
+	/**
+	 * Get API version to use for some of the Controller's API route mappings based on the kubernetes
+	 * version used in the underlying cluster.
+	 * @param client the KubernetesHttpClient to use
+	 * @return the appropriate API version to use
+	 */
+	public static String getApiVersionForK8sVersionFromCluster(KubernetesHttpClient client) {
+		try {
+			Response response = client.get("version", "");
+			Map<String, String> versionResponse = new ObjectMapper().readValue(response.body().string(), HashMap.class);
```

Comment:
the unchecked assignment can be avoided by using say `new TypeReference<HashMap<String, String>>(){}` rather than `HashMap.class`.